curriculum/challenges/english/20-upcoming-python/learn-list-comprehension-by-building-a-case-converter-program/657f456223b8c1187b461987.md

```diff
@@ -0,0 +1,64 @@
+---
+id: 657f456223b8c1187b461987
+title: Step 18
+challengeType: 20
+dashedName: step-18
+---
+
+# --description--
+
+When you start a list comprehension with an `if` statement like this, Python requires you to also add an `else` statement to the expression.
+
+```py
+snake_cased_char_list = [
+    '_' + char.lower() if char.isupper()
+    else char
+]
+```
+
+Python will interpret this updated expression as "append `'_' + char.lower()` to the list if `char` is in uppercase, append `char` as is otherwise" and this covers the case for both the capital and small letters in the input string.
```

Comment:
```suggestion
Python will interpret this updated expression as "append `'_' + char.lower()` to the list if `char` is in uppercase, append `char` as is otherwise" and this covers the case for both the capital and lowercase letters in the input string.
```